src/hooks/useBifrostWebSocketEffect.ts

```diff
@@ -50,40 +66,43 @@ interface IUseBifrostWebSocketEffectOptions extends IUsePncWebSocketEffectOption
  */
 export const useBifrostWebSocketEffect = (
   callback: (logLine: string) => void,
-  { preventListening = false, debug = '', buildId = '' }: IUseBifrostWebSocketEffectOptions = {}
+  { preventListening = false, debug = '', buildId, closeResultId }: IUseBifrostWebSocketEffectOptions = {}
 ) => {
   useEffect(() => {
-    if (!preventListening) {
-      if (!buildId) {
-        uiLogger.error('Missing buildId for useBifrostWebSocketEffect');
-        return;
-      }
-      const prefixFilters = buildId ? buildLogPrefixFilters : null;
-      const matchFiltersPrefix = buildId ? buildLogMatchFiltersPrefix : null;
-      // Send subscribe message
-      const subscribeMessage = {
-        jsonrpc: '2.0',
-        id: 0,
-        method: 'SUBSCRIBE',
-        params: {
-          prefixFilters: prefixFilters,
-          matchFilters: matchFiltersPrefix + buildId,
-        },
-      };
-
-      bifrostWebSocketClient.sendMessage(JSON.stringify(subscribeMessage));
-
-      const onMessage = (wsMessage: MessageEvent) => {
-        const wsResponseData: IWsResponseData = JSON.parse(wsMessage.data);
-        const resultValue = wsResponseData.result?.value;
-        resultValue?.message && callback(`[${resultValue.timestamp}] ${resultValue.message}`);
-      };
-
-      const removeMessageListener = bifrostWebSocketClient.addMessageListener(onMessage, { debug });
-
-      return () => {
-        removeMessageListener();
-      };
+    if (preventListening) return;
+
+    if (!(buildId || closeResultId)) {
+      uiLogger.error('Missing buildId or closeResultId for useBifrostWebSocketEffect');
+      return;
     }
-  }, [callback, buildId, preventListening, debug]);
+
+    const prefixFilters = getPrefixFilters(buildId, closeResultId);
+    const matchFilters = getMatchFilters(buildId, closeResultId);
```

Comment:
This was refactored because we also need to consider other live logs like brew push log that have 2 params for matchFilters